wdns/rdata_to_ubuf.c

```diff
@@ -503,6 +547,15 @@ _wdns_rdata_to_ubuf(ubuf *u, const uint8_t *rdata, uint16_t rdlen,
 					goto err;
 				}
 
+				if (key == spr_invalid) {
+					goto err;
+				}
+
+				/* param keys must be in ascending order */
```

Comment:
While the draft spec requires keys to be sorted, for passive DNS purposes we should not reject these -- we can present the parameters in the order they appear on the wire (and conversely parse that presentation back to the original wire layout), and should.